### DISCUSSION TITLE
Looking for advice on error - Token for '%s' specifier is NULL. (Sorry, opened an issue first by mistake)

### DISCUSSION THREAD

**Discussion by avonnieda** [allinurl/goaccess#2674] [Q&A] [answered]
Hi, looking for advice on this error - Token for '%s' specifier is NULL.
Thank you for any insight you can provide, very excited to get this working!

goaccess config
time-format %H:%M:%S
date-format %d/%b/%Y
log-format %h %{%s}t %l %u %t "%r" %>s %b %T/%D

Apache config
LogFormat "%h %{%s}t %l %u %t "%r" %>s %b %T/%D" response_times
CustomLog logs/ssl_response_log response_times

Sample log entries
192.168.1.5 1713687672 - - [21/Apr/2024:03:21:12 -0500] "GET /foo/bar/bat/service?p_1=036&p_2=ALL&p_3=ALL&p_4=ALL HTTP/1.1" 200 10286 22/22923423
192.168.1.5 1713687696 - - [21/Apr/2024:03:21:36 -0500] "GET /foo/bar/bat/service?p_1=039&p_2=ALL&p_3=ALL&p_4=ALL HTTP/1.1" 200 97772 18/18612738

Full error
==2083864== Parsed 10 lines producing the following errors:
==2083864==
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864== Token for '%s' specifier is NULL.
==2083864==
==2083864== Format Errors - Verify your log/date/time format

**Comment by allinurl** [comment, marked as answer]
This should do it, let me know how it goes:

    goaccess access.log --log-format='%h %^[%d:%t %^] "%r" %s %b %^/%D' --date-format=%d/%b/%Y --time-format=%T --ignore-panel=BROWSERS --ignore-panel=OS --ignore-panel=REFERRING_SITES --date-spec=min

![2024-04-25-105312_599x289_scrot](https://github.com/allinurl/goaccess/assets/5005367/4ec53e3b-6371-457b-893c-f34086e64895)

**Reply by avonnieda** [reply]
Holy smokes that was fast, and yep that did it!! Thank you so much!!

Edit: This software is awesome!

**Reply by allinurl** [reply]
Happy to hear that did the job. Don't hesitate to reopen it if necessary.
